scripts/_common.sh

```diff
@@ -53,222 +55,250 @@ exec_as() {
-			logfile=$1	# In this case, focus logrotate on the logfile
-		else
-			logfile=$1/.log	# Else, uses the directory and all logfile into it.
-		fi
-	else
-		logfile="/var/log/${app}/.log" # Without argument, use a defaut directory in /var/log
-	fi
-	cat > ./${app}-logrotate << EOF	# Build a config file for logrotate
-$logfile {
-		# Rotate if the logfile exceeds 100Mo
-	size 100M
-		# Keep 12 old log maximum
-	rotate 12
-		# Compress the logs with gzip
-	compress
-		# Compress the log at the next cycle. So keep always 2 non compressed logs
-	delaycompress
-		# Copy and truncate the log to allow to continue write on it. Instead of move the log.
-	copytruncate
-		# Do not do an error if the log is missing
-	missingok
-		# Not rotate if the log is empty
-	notifempty
-		# Keep old logs in the same dir
-	noolddir
-}
-EOF
-	sudo mkdir -p $(dirname "$logfile")	# Create the log directory, if not exist
-	cat ${app}-logrotate | sudo tee -a /etc/logrotate.d/$app > /dev/null	# Append this config to the others for this app. If a config file already exist
-}
-
-# Remove the app's logrotate config.
+# To specify which files it has to move, you have to create a "migration file", stored in ../conf
+# This file is a simple list of each file it has to move,
+# except that file names must reference the $app variable instead of the real name of the app,
+# and every instance-specific variables (like $domain).
+# $app is especially important because it's this variable which will be used to identify the old place and the new one for each file.
 #
-# usage: ynh_remove_logrotate
-ynh_remove_logrotate () {
-	if [ -e "/etc/logrotate.d/$app" ]; then
-		sudo rm "/etc/logrotate.d/$app"
-	fi
-}
-
-# Calculate and store a file checksum into the app settings
+# If a database exists for this app, it will be dumped and then imported in a newly created database, with a new name and new user.
+# Don't forget you have to then apply these changes to application-specific settings (depends on the packaged application)
 #
-# $app should be defined when calling this helper
+# Same things for an existing user, a new one will be created.
+# But the old one can't be removed unless it's not used. See below.
 #
-# usage: ynh_store_file_checksum file
-# | arg: file - The file on which the checksum will performed, then stored.
-ynh_store_file_checksum () {
-	local checksum_setting_name=checksum_${1//[\/ ]/_}	# Replace all '/' and ' ' by '_'
-	ynh_app_setting_set $app $checksum_setting_name $(sudo md5sum "$1" | cut -d' ' -f1)
-}
-
-# Verify the checksum and backup the file if it's different
-# This helper is primarily meant to allow to easily backup personalised/manually 
-# modified config files.
+# If you have some dependencies for your app, it's possible to change the fake debian package which manages them.
+# You have to fill the $dependencies variable, and then a new fake package will be created and installed,
```

Comment:
Many apps use `$pkg_dependencies`, shouldn't we use that name instead?

Reply:
As you want :)